website/sales/api/v2/serializers/order.py

```diff
@@ -0,0 +1,177 @@
+            "payment",
+            "payment_amount",
+            "total_amount",
+            "age_restricted",
+            "order_description",
+            "payment_url",
+        )
+
+    shift = serializers.HyperlinkedRelatedField(
+        view_name="v2:shift-detail", read_only=True
+    )
+
+    order_items = OrderItemSerializer(many=True, required=False)
+
+    total_amount = serializers.DecimalField(
+        max_digits=6, decimal_places=2, read_only=True
+    )
+    payment_amount = serializers.DecimalField(
+        max_digits=6, decimal_places=2, read_only=True
+    )
+
+    payment = serializers.HyperlinkedRelatedField(
+        view_name="v1:payment-detail", read_only=True
+    )  # TODO bump version to v2 when available
+
+    payer = serializers.HyperlinkedRelatedField(
+        view_name="v2:member-detail", read_only=True
+    )  # TODO writing to this causes errors
+
+    payment_url = serializers.URLField(read_only=True)
+
+    def create(self, validated_data):
+        shift = self.context["shift"]
+        order = Order.objects.create(shift=shift)
+        if "order_items" in validated_data:
+            items_data = validated_data.pop("order_items")
+            for item_data in items_data:
+                item_data["order"] = order
+                OrderItem.objects.create(order=order, **item_data)
+        return order
+
+    def update(self, instance, validated_data):
+
+        # Update the order items for an order
+        if "order_items" in validated_data:
+            items_data = validated_data.pop("order_items")
+            current_items = list(instance.order_items.all())
+
+            # Overwrite all existing order items by the newly provided ones
+            for item_data in items_data:
+                if len(current_items) > 0:
+                    item = current_items.pop(0)
+                    OrderItemSerializer(item, context={"order": instance}).update(
+                        item, item_data
+                    )
+                else:
+                    # Create new order items if required
+                    OrderItemSerializer(context={"order": instance}).create(
+                        validated_data=item_data
+                    )
+
+            # Delete all order items that we have not updated
+            for i in current_items:
+                i.delete()
+
+        # Update other fields of the order as default
+        super().update(instance, validated_data)
+        return instance
+
+
+class OrderListSerializer(OrderSerializer):
+    class Meta:
+        model = Order
+        fields = (
+            "pk",
+            "created_at",
+            "payment_amount",
+            "payment",
+            "payer",
+            "details",
+        )
+        read_only_fields = (
+            "pk",
+            "created_at",
+            "payment_amount",
+            "payment",
+            "payer",
+            "details",
+        )
+
+    payer = serializers.HyperlinkedRelatedField(
```

Comment:
This could just be a `MemberSerializer` to immediately include the information. You basically already know you are going to need it.

Reply:
Maybe in a smaller form but that would be pretty easy to do